src/main/scala/com/vngrs/etl/utils/rdd.scala

```diff
@@ -0,0 +1,58 @@
+package com.vngrs.etl.utils
+
+import com.vngrs.etl.exceptions.NoSchemaException
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.types.StructType
+
+import scala.util.Try
+
+/**
+  * Enriches [[org.apache.spark.rdd.RDD]] with more functionality
+  */
+object rdd {
+
+  /**
+    * Enriches [[org.apache.spark.rdd.RDD]] to have more functionality
+    *
+    * @param rdd [[org.apache.spark.rdd.RDD]]
+    * @tparam T Type of the RDD
+    */
+  implicit final class RichRDD[T](val rdd: RDD[T]) extends AnyVal {
+
+    /**
+      * Checks whether RDD is non empty or not
+      * @return
+      */
+    def nonEmpty(): Boolean = !rdd.isEmpty()
+  }
+
+  /**
+    * Enriches [[org.apache.spark.rdd.RDD]] of [[org.apache.spark.sql.Row]]s
+    * to have more functionality
+    *
+    * @param rdd [[org.apache.spark.rdd.RDD]]
+    */
+  implicit final class RichRowRDD(val rdd: RDD[Row]) extends AnyVal {
+
+    /**
+      * Gets Schema of the first [[org.apache.spark.sql.Row]] as [[scala.Option]]
+      *
+      * @return Schema of the rows if rdd is non empty and rows have a schema, otherwise `None`
+      */
+    def schemaOpt: Option[StructType] = Try(rdd.first()).toOption.flatMap(r => Option(r.schema))
+
+    /**
+      * Gets Schema of the first [[org.apache.spark.sql.Row]] as [[scala.Option]]
+      * or throws [[com.vngrs.etl.exceptions.NoSchemaException]].
+      *
+      * @return Schema of the rows if rdd is non empty and rows have a schema
+      * @throws com.vngrs.etl.exceptions.NoSchemaException if no schema is found
+      */
+    @throws(classOf[NoSchemaException])
+    def schemaOrThrow: StructType = schemaOpt match {
+      case Some(s)  => s
+      case None     => throw new NoSchemaException("A Schema expected to be found in Row bu none is found!")
```

Comment:
Exception message should be customizable to provide better information about failure to users